### PR TITLE
DASHBOARD: Add github workflow used to build the karmada-dashboard image

### PR DESCRIPTION
**Pull request by devadapter**
Fixes issue: [dashboard#4](https://github.com/karmada-io/dashboard/issues/4)

### CODE REVIEW

File: .github/workflows/push-image.yml [hunk 1]
```diff
@@ -0,0 +1,37 @@
+name: Image-Build-And-Push
+
+on:
+  push:
+    branches: [main]
+  pull_request:
+    branches: [main]
```
Comment (RainbowMango) [inline .github/workflows/push-image.yml:7]:
```suggestion
```
We shouldn't publish images on PR since the PR hasn't gotten approved.

Comment (RainbowMango) [reply .github/workflows/push-image.yml:7]: We will publish images only after PR gets merged to `main` branch.

File: .github/workflows/push-image.yml [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+
+jobs:
+  build-and-push:
```
Comment (RainbowMango) [inline .github/workflows/push-image.yml:10]: We should stop the jobs from `forked` repository, and the reason just like what we did on [Karmada repo](https://github.com/karmada-io/karmada/blob/808e487a2d977e9be9c3d38747d3f11f01826146/.github/workflows/image.yml#L9-L12).

**Comment by RainbowMango** [comment]
@devadapter can we add a `makefile` and users can run `make images` to build images then `make upload-images` to any registry they specified. 

We can refer to Karmada Makefile:
https://github.com/karmada-io/karmada/blob/master/Makefile
And the workflow:
https://github.com/karmada-io/karmada/blob/master/.github/workflows/image.yml

File: .github/workflows/push-image.yml (hunk 2 above)
Comment (devadapter) [reply .github/workflows/push-image.yml:10]: Thanks, I will revise it.

**Comment by devadapter** [comment]
> @devadapter can we add a `makefile` and users can run `make images` to build images then `make upload-images` to any registry they specified.
> 
> We can refer to Karmada Makefile: https://github.com/karmada-io/karmada/blob/master/Makefile And the workflow: https://github.com/karmada-io/karmada/blob/master/.github/workflows/image.yml

This workflow pushes images to different registry based on secret variables, if you need to switch to Makefile, I'll write about it later as I'm not familiar with it

**Comment by RainbowMango** [comment]
> This workflow pushes images to different registry based on secret variables, if you need to switch to Makefile, I'll write about it later as I'm not familiar with it

OK.

**Comment by RainbowMango** [comment]
I'm going to merge this workflow first.

**Pull request merged**